### PR TITLE
Documentation examples improvements

### PR DESCRIPTION
**Pull request by Gr1N**
Hi,

Some my ideas and improvements for examples in docstrings.

### CODE REVIEW

File: paco/concurrent.py [hunk 1]
```diff
@@ -92,11 +92,17 @@ class ConcurrentExecutor(object):
...
+        async def sum(x, y):
+            return x + y
+
+        pool = paco.ConcurrentExecutor(limit=2)
+        pool.add(sum, 1, 2)
+        pool.add(sum, None, 'str')
+
+        done, pending = await pool.run(return_exceptions=True)
+        [task.result() for task in done]
+        # => [3, TypeError("unsupported operand type(s) for +: 'NoneType' and 'str'")]
```
Comment (h2non) [inline paco/concurrent.py]: This line exceeds the maximum limit. We should add `# noqa` at the end of the line, such as:
```python
# => [3, TypeError("unsupported operand type(s) for +: 'NoneType' and 'str'")]  # noqa
```

File: paco/gather.py [hunk 2]
```diff
@@ -45,12 +45,15 @@ def gather(*coros_or_futures, limit=0, loop=None, timeout=None,
...
         await paco.gather(
-          task(1, foo='bar'),
-          task(2, foo='bar'),
-          task(3, foo='bar'),
-          task(4, foo='bar'),
-          limit=2, return_exceptions=True)
+            sum(1, 2),
+            sum(None, 'str'),
+            return_exceptions=True)
+        # => [3, TypeError("unsupported operand type(s) for +: 'NoneType' and 'str'")]
```
Comment (h2non) [inline paco/gather.py]: Same here:
```python
[3, TypeError("unsupported operand type(s) for +: 'NoneType' and 'str'")]  # noqa
```

**Comment by Gr1N** [comment]
Linter issues fixed.

About `doctest`, I can try to rewrite all examples in `doctest` style, but as I understand there some limitations, e.g. we can't use `await` syntax.

Some example according to `doctest` format:
```
>>> async def mul_2(num):
...     return num * 2
...
>>> paco.run(paco.map(mul_2, [1, 2, 3, 4, 5]))
[2, 4, 6, 8, 10]
```

**Comment by h2non** [comment]
Yes, that would be a bit tricky... I think we can consider that in another PR / enhancement iteration.

Is it fine for you if I proceed with the merge?

**Comment by h2non** [comment]
You can also create a new section in the README called Contributors and add there yourself :)

**Comment by Gr1N** [comment]
> Is it fine for you if I proceed with the merge?

Yes, if PR looks good for you.

> You can also create a new section in the README called Contributors and add there yourself :)

Thanks, but I think if anyone want to see list of contributors he can just run:
```
% git log --format='%aN' | sort -u
```
:)

**Comment by tomas-fp** [comment]
re contributors, in that case I'd prefer using `git-extras` instead:
https://github.com/tj/git-extras/blob/master/Commands.md#git-summary

**Pull request merged**